src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,190 @@
+
+    std::string examinedDigraphs{""};
+
+    // If repeated chars in a digraph add an X or Q if XX
+    // Look at each character in inputText
+    // If the character before is the same as the current one, replace with X + current character
+    // If both the characters are XX, replace with QX.
+    for (std::size_t i{0}; i < inputText.size(); i++){
+        // only doing something on every second letter
+        if (i == 1 % 2){
+            // checking if there are pairs of repeated letters
+            if (inputText[i] == inputText[i-1]){
+                // if the repeated letters are XX, populate examinedDigrahs with QX
+                if (inputText[i] == 'X'){
+                    examinedDigraphs+= "QX";
+                    
+                }
+                // if the repeated letters are not XX, populate examinedDigraphs with X(letter)
+                else{
+                    examinedDigraphs+= "X";
+                    examinedDigraphs+= inputText[i];
+                    
+
+                }
+            }
+            // if the pair are not the same letter, populate examinedDigraphs with the letters
+            else{
+                examinedDigraphs+= inputText[i-1];
+                examinedDigraphs+= inputText[i];
+                
+            }
+
+
+        }
+    }
+
+
+    // If the length of inputText is odd, then the above loop did not add the last letter to examinedDigraphs.
+    // We need to add this letter and a Z to examinedDigraphs
+    // if the size of input is odd, add a trailing Z
+    if (inputText.size() == 1 % 2){
+        examinedDigraphs+= inputText[inputText.size()-1];
+        examinedDigraphs+= "Z";
+    }
+
+    // Comments about my code to be looked at whilst giving feedback:
+    // I am aware that my code above would not put the letters into the right pairs. 
+    // HELLO WORLD would become HE XL OW OR DL
+    // when it should become HE LX LO WO RL DZ
+    // But i can't find a way to get the splitting into digraphs right every single time. 
+    // When you add the X to separate repeated letters, you add an odd number of letters 
+    // to the string and therefore you change what letters are paired together 
+    // E.g. if you were to pair HELLO W before consiering repeated letters you would get HE LL OW
+    // But when you add the x to separate the Ls you get HE LX LO W.....
+    // I'm not sure how to combat this with a loop that uses indices. 
+    // I also think I need some clrification about how teh cipher works
+    // Do we separate any repeated letters with an X before putting the string into digraphs?
+    // e.g HEEL becomes HEXEL becomes HE XE LZ.
+    // Or do we first split into digraphs and then add the Xs if the repeating letters are in the same digraph 
+    // and therefore change the digraphs at this point
+    // e.g HEEL becomes HE EL and there is no problem with that because the Es aren't in the same digraph.
```

Comment:
Hopefully the above comments answer these questions but just to be sure I'll give explicit answers here as well.

The answer to the first part is that you have to loop in jumps of 2, examining elements `[i]` and `[i+1]`, and when you encounter a double letter in a digraph you don't add the second character but add something different instead and (most importantly) you must then do a decrement of the loop counter by 1 so that the second element in that digraph becomes the first in the next.

For the second part, HEEL should become HE EL - there is no need to insert an X here because the repetition does not occur within a digraph.  The reason behind this is that if you have two identical characters in a digraph, the encryption/decryption rules become ambiguous since they are in the same row and in the same column and form a rectangle, so which rule should you apply?  So to avoid this problem, digraphs are not allowed to have repeated letters.  However, in the HEEL case the repeat is not within a digraph, so there's no problem and hence no need to add any extra letters.